src/res_core.ml

```diff
@@ -665,39 +668,42 @@ let parseIdent ~msg ~startPos p =
       "`" ^ tokenTxt ^ "` is a reserved keyword. Keywords need to be escaped: \\\"" ^ tokenTxt ^ "\""
     in
     Parser.err ~startPos p (Diagnostics.message msg);
-    Parser.next p;
     (tokenTxt, mkLoc startPos p.prevEndPos)
   | _token ->
     Parser.err ~startPos p (Diagnostics.message msg);
-    Parser.next p;
     ("", mkLoc startPos p.prevEndPos)
 
+let parseIdent ~msg ~startPos p =
+  let res = parseIdentNoNext ~msg ~startPos p in
+  Parser.nextUnsafe p;
```

Comment:
When we don't know, just do `nextUnsafe`. That's OK as long as termination analysis is happy.